lib/client-bundle-plugin.js

```diff
@@ -20,18 +20,31 @@ const allSrcPlaceholder = 'CLIENT_BUNDLE_PLUGIN_ALL_SRC';
 
 export function getDependencies(clientOutput, item) {
   const crawlDependencies = new Set([item.fileName]);
+  const referencedFiles = new Set();
 
   for (const fileName of crawlDependencies) {
     const chunk = clientOutput.find((v) => v.fileName === fileName);
 
     for (const dep of chunk.imports) {
       crawlDependencies.add(dep);
     }
+
+    for (const dep of chunk.referencedFiles) {
+      referencedFiles.add(dep);
+    }
   }
 
   // Don't add self as dependency
   crawlDependencies.delete(item.fileName);
 
+  // Merge referencedFiles as regular deps.
+  //
+  // Didn't do this as part of the main loop since their `chunk` can't have
+  // nested deps and sometimes might be missing altogether, depending on type.
+  for (const dep of referencedFiles) {
+    crawlDependencies.add(dep);
+  }
+
   return [...crawlDependencies];
```

Comment:
Seems a bit weird to add referenced files to `crawlDependencies`. Does this work?

```suggestion
  return [...crawlDependencies, ...referencedFiles];
```

Reply:
Not really, because some JS can be simultaneously a regular entry/dependency and a referenced file (like auto-generated Wasm workers). I could create another intermediate `Set` if you want though.

Reply:
Sticking with the current code but adding a comment making the above clear seems fine

Reply:
```suggestion
  // Merge referencedFiles as regular deps. They need to be in the same Set as
  // some JS files might appear in both lists and need to be deduped too.
  //
  // Didn't do this as part of the main loop since their `chunk` can't have
  // nested deps and sometimes might be missing altogether, depending on type.
  for (const dep of referencedFiles) {
    crawlDependencies.add(dep);
  }

  return [...crawlDependencies];
```